### PR TITLE
Add F-engine `rx.device-status` sensor.

### PR DESCRIPTION
**Pull request by james-smith-za**
Turned out to be relatively simple.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update ``setup.cfg`` and appropriate requirements files
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in ``doc/``
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match
(See https://github.com/ska-sa/katsdpcontroller/pull/630)
Closes NGC-847.

### CODE REVIEW

**Comment by james-smith-za** [comment]
I have to get on the bike in a minute, I'll do the corresponding thing in `katsdpcontroller` tomorrow probably.

**Review by amishatishpatel — APPROVED** [review]
I assume we're going on the ICD draft description of sensors. Either way, big fan.

File: src/katgpucbf/fgpu/recv.py [hunk 1]
```diff
@@ -255,6 +255,11 @@ def make_sensors(sensor_timeout: float) -> aiokatcp.SensorSet:
         for sensor in missing_sensors:
             TimeoutSensorStatusObserver(sensor, sensor_timeout, aiokatcp.Sensor.Status.NOMINAL)
             sensors.add(sensor)
+
+        sensors.add(
+            DeviceStatusSensor(sensors, "rx.device-status", "F-engine is receiving a good, clean digitiser stream")
+        )
```
Comment (bmerry) [inline src/katgpucbf/fgpu/recv.py:261]: Should be outside the loop over `pol` - currently you're making two of these sensors (one of which will just be garbage-collected).

**Review by bmerry — CHANGES_REQUESTED** [review]
We still need to resolve where we're using `.` versus `-` in sensor names: it's currently inconsistent, with existing sensors starting `rx-` and the new one starting `rx.`. That can be a separate PR I guess.

**Comment by bmerry** [comment]
The ICD specifies this as a boolean sensor, not a discrete.

**Comment by amishatishpatel** [comment]
> The ICD specifies this as a boolean sensor, not a discrete.

Ah damn, I think my coffee hadn't fully kicked in. I had both the old and new ICDs up looking for commonalities and finding the description, failed to pay enough attention to the Type - apologies.

File: src/katgpucbf/fgpu/recv.py (hunk 1 above)
Comment (james-smith-za) [reply src/katgpucbf/fgpu/recv.py:261]: Good catch!

**Comment by bmerry** [comment]
> The ICD specifies this as a boolean sensor, not a discrete.

It looks like the MeerKAT *.device-status sensors are all discrete though. That makes me think the MK+ ICD should perhaps be changed.

**Comment by james-smith-za** [comment]
> > The ICD specifies this as a boolean sensor, not a discrete.
> 
> It looks like the MeerKAT *.device-status sensors are all discrete though. That makes me think the MK+ ICD should perhaps be changed.

On that point, it sort of makes me think that we should use the `rx.something` pattern for the other sensors, making it clear that they're in the same logical unit that's feeding into `rx.device-status`.

**Comment by james-smith-za** [comment]
> We still need to resolve where we're using `.` versus `-` in sensor names: it's currently inconsistent, with existing sensors starting `rx-` and the new one starting `rx.`. That can be a separate PR I guess.

NGC-854

**Review by bmerry — APPROVED** [review]

File: src/katgpucbf/fgpu/recv.py [hunk 2]
```diff
@@ -256,9 +256,7 @@ def make_sensors(sensor_timeout: float) -> aiokatcp.SensorSet:
             TimeoutSensorStatusObserver(sensor, sensor_timeout, aiokatcp.Sensor.Status.NOMINAL)
             sensors.add(sensor)
 
-        sensors.add(
-            DeviceStatusSensor(sensors, "rx.device-status", "F-engine is receiving a good, clean digitiser stream")
-        )
```
Comment (amishatishpatel) [inline src/katgpucbf/fgpu/recv.py]: The one time I view the PR in GitHub and not in a text editor is obviously the one time an actual change needed to be addressed. Please and thank yous all around.

**Review by amishatishpatel — APPROVED** [review]

**Pull request merged**